tests/cross/src/test/scala/tests/hover/HoverTermSuite.scala

```diff
@@ -210,7 +217,10 @@ class HoverTermSuite extends BaseHoverSuite {
       |  }
       |}
       |""".stripMargin,
-    "class Foo: Foo".hover
+    "class Foo: Foo".hover,
+    compat = Map(
+      "3" -> "def <init>(name: String, age: Int): Foo".hover
```

Comment:
For compatibility between Scala 2 and Scala 3 we should most likely show the Foo type, since this is not actually the constructor but type definition.

Reply:
Fixed by https://github.com/scalameta/metals/pull/3854/commits/287235e39c6e68b0dea83fa0037b46afc34ebd66
but I'm wondering maybe we should go with `def <init>(name: String, age: Int): Foo`, at least in Scala3 (and in Scala2 too if possible)? 🤔 
Even though there's indeed anonymous class definition under the hood, showing constructor seems more meaningful.

Reply:
Hmm... this might be a limitation of the Scala 2 implementation then. It's ok to show the constructor here.

Reply:
Ok, I reverted https://github.com/scalameta/metals/commit/287235e39c6e68b0dea83fa0037b46afc34ebd66 in https://github.com/scalameta/metals/pull/3854/commits/ed1d5287c91f820762089c2333d8ce11607ae3ca :+1: